adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -67,27 +70,27 @@ - (void)requestRewardedAd {
   }
   
   if (![UnityAds isInitialized]) {
-    [[GADMAdapterUnity alloc] initializeWithGameID:_gameID withInitDelegate:Nil];
+    [[GADMAdapterUnity alloc] initializeWithGameID:_gameID withCompletionHandler:nil];
   }
 
-  @synchronized (_placementInUseRewarded) {
+  dispatch_sync(_lockQueue, ^{
     if ([_placementInUseRewarded objectForKey:_placementID]) {
       if (_adLoadCompletionHandler) {
         NSError *error = GADUnityErrorWithDescription([NSString stringWithFormat:@"An ad is already loading for placement ID: %@.", _placementID]);
         _adEventDelegate = _adLoadCompletionHandler(nil, error);
       }
       return;
     }
-    [_placementInUseRewarded setValue:self forKey:_placementID];
-  }
+    GADMAdapterUnityMapTableSetObjectForKey(_placementInUseRewarded, _placementID, self);
+  });
 
   [UnityAds load:_placementID loadDelegate:self];
 }
 
 - (void)presentFromViewController:(nonnull UIViewController *)viewController {
-  @synchronized (_placementInUseRewarded) {
-    [_placementInUseRewarded removeObjectForKey:_placementID];
-  }
+  dispatch_sync(_lockQueue, ^{
```

Comment:
Per https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/CONTRIBUTING.md#best-practices, writes can be asynchronous. Only reads need to be synchronous, which will block if any writes haven't been written.